docs/csharp/whats-new/tutorials/primary-constructors.md

```diff
@@ -0,0 +1,102 @@
+
+The most common uses for a primary constructor parameter are:
+
+1. As an argument to a `base()` constructor invocation.
+1. To initialize a member field or property.
+1. Referencing the constructor parameter in an instance member.
+
+Every other constructor for a class **must** call the primary constructor, directly or indirectly, through a `this()` constructor invocation. That rule ensures that primary constructor parameters are assigned anywhere in the body of the class.
+
+## Initialize property
+
+The following code initializes two readonly properties that are computed from primary constructor parameters:
+
+:::code source="./snippets/primary-constructors/Distance.cs" id="ReadonlyStruct":::
+
+The preceding code demonstrates a primary constructor used to initialize calculated readonly properties. The field initializers for `Magnitude` and `Direction` use the primary constructor parameters. The primary constructor parameters aren't used anywhere else in the struct. The preceding struct is as though you'd written the following code:
+
+:::code source="./snippets/primary-constructors/Distance.cs" id="StructOneLowered":::
+
+The new feature makes it easier to use field initializers when you need arguments to initialize a field or property.
+
+## Create mutable state
+
+The preceding examples use primary constructor parameters to initialize readonly properties. You can also use primary constructors when the properties aren't readonly. Consider the following code:
+
+:::code source="./snippets/primary-constructors/Distance.cs" id="MutableStruct":::
+
+In the preceding example, the `Translate` method changes the `dx` and `dy` components. That requires the `Magnitude` and `Direction` properties be computed when accessed. The `=>` operator designates an expression-bodied `get` accessor, whereas the `=` operator designates an initializer. This version adds a parameterless constructor to the struct. The parameterless constructor must invoke the primary constructor, so that all the primary constructor parameters are initialized.
+
+In the previous example, the primary constructor properties are accessed in a method. Therefore the compiler creates hidden fields to represent each parameter. The following code shows approximately what the compiler generates. The actual field names are valid MSIL identifiers, but not valid C# identifiers.
+
+:::code source="./snippets/primary-constructors/Distance.cs" id="StructTwoLowered":::
+
+It's important to understand that the first example didn't require the compiler to create a field to store the value of the primary constructor parameters. The second example used the primary constructor parameter inside a method, and therefore required the compiler create storage for them. The compiler creates storage for any primary constructors only when that parameter is accessed in the body of a member of your type. Otherwise, the primary constructor parameters aren't stored in the object.
+
+## Dependency injection
```

Comment:
Maybe, that should be the first in the list of examples? The preceding two may require argument validation.

Reply:
I like the idea of adding argument validation. I've left the order as is. I think it flows better to have a simpler example first. (And, any double *should* be valid for a vector).